htlcswitch/interceptable_switch.go

```diff
@@ -139,7 +139,16 @@ func (f *interceptedForward) Resume() error {
 
 // Fail forward a failed packet to the switch.
 func (f *interceptedForward) Fail() error {
-	reason, err := f.packet.obfuscator.EncryptFirstHop(lnwire.NewTemporaryChannelFailure(nil))
+	update, err := f.htlcSwitch.cfg.FetchLastChannelUpdate(
+		f.packet.incomingChanID,
+	)
+	if err != nil {
+		return err
```

Comment:
If from some reason this returns an error (like when channel edge is not found) the user won't be able to fail the intercepted HTLC.
I wonder if in this case it is better to pick a different reason for the UpdateFailHTLC, maybe a reason that doesn't require channel update (lnwire.FailTemporaryNodeFailure for example).

Reply:
If the HTLC arrived on the incoming edge, I can't think of a practical reason that this call would fail, we need to fetch this policy in order to validate the normal HTLC forwarding information. 

See the prior PR for the rationale re why using a node level failure would result in degraded path finding: https://github.com/lightningnetwork/lnd/pull/5059#issuecomment-788375918

Reply:
@roeierez would a possible approach for this be to: 
1. Proceed with this change, so that the issue with eclair is fixed
2. Follow up with a PR which will reject the HTLC by default if we experience an unexpected error during interception? Seems like reasonable behaviour to me if we're worried about not being able to intercept.

Reply:
@carlaKC yes sounds good.